### PR TITLE
Fix CIS state and review comments

### PR DESCRIPTION
**Pull request by leodotcloud**
Addresses: https://github.com/rancher/rancher/issues/25214
Addresses comments from: https://github.com/rancher/rancher/pull/25475

### CODE REVIEW

File: pkg/api/customization/clusterscan/clusterscan.go [hunk 1]
```diff
@@ -33,11 +40,28 @@ func Formatter(apiContext *types.APIContext, resource *types.RawResource) {
...
+			if !ok {
+				resource.Values["state"] = errorState
+				return
+			}
+			cisScanStatus := convert.ToMapInterface(cisScanStatusInterface)
+			failInterface := cisScanStatus["fail"]
+			fail, err := convert.ToNumber(failInterface)
+			if err != nil {
+				logrus.Errorf("error converting failInterface to int: %v", err)
+				resource.Values["state"] = errorState
```
Comment (luthermonson) [inline pkg/api/customization/clusterscan/clusterscan.go]: should you return after this? if err is != nil likely fail == 0 meaning the the else would mark it passed right?

Comment (leodotcloud) [reply pkg/api/customization/clusterscan/clusterscan.go]: yup, let me fix it.

File: pkg/controllers/user/cis/utils.go [hunk 2]
```diff
@@ -32,22 +32,14 @@ func createConfigMapWithRetry(configMapsClient rcorev1.ConfigMapInterface, cm *v
...
-		)
-		if err != nil {
-			return fmt.Errorf("error listing pods: %v", err)
-		}
-		if len(pods) == 0 {
-			removed = true
-			break
-		}
-		time.Sleep(RetryIntervalInMilliseconds * time.Millisecond)
+	pods, err := podLister.List(
```
Comment (dramich) [inline pkg/controllers/user/cis/utils.go]: We should prefer to use indexes vs a lister with a selector. The lister will loop over all objects to build your return and on something that will most likely have a lot of things it's resource intensive.

Comment (leodotcloud) [reply pkg/controllers/user/cis/utils.go]: This is an ephemeral pod so not sure if having an index would be of any benefit. Also, there is only one pod running in a cluster with this particular label set.

**Pull request merged**